chore: reserved query parameter `has_permission` (#294)
snyk/sweater-comb: docs/standards/rest.md

[Header field names are case insensitive](https://datatracker.ietf.org/doc/html/rfc7230#section-3.2). Snyk v3 API specs must use kebab case for consistency. All non-standard headers that are unique to Snyk must begin with `snyk-` (e.g. `snyk-requested-version`).

## <a id="reserved-request-parameters"></a>Reserved Request Parameters

### <a id="filtering-authorized-access"></a>Filtering authorized access

Some resources may allow filtering results based on authorizations granted to the authenticated principal. Some applications of this:

- API applications: "Apply a policy to those resources the caller is an administrator of."
- User interfaces: "Display editing widgets for those resources the user can edit."

> #### `has_permission` query parameter

Author: We might have a common schema type that defines these. However, it's also possible that some permissions are not relevant to some resources. In that case it might be reasonable for an endpoint to only declare a subset of those permissions.

> #### `has_permission` query parameter

Reviewer: I think a convention for naming the slugs would be good enough, eg `org.read` -> `org_read`.

> A URL query parameter `?has_permission=permission_slug` may be used on a collection to filter results by permissions authorized to the authenticated principal. When this parameter is present, the schema for `permission_slug` must be an enum type (`{"type": "string", "enum": [...]}`).

Reviewer 2: ```suggestion
A URL query parameter `?has_permission=permission_slug` may be used on a collection to filter results by permissions authorized to the authenticated principal. 

The parameter is a list of `permission_slug` eg: `?has_permission=one,two`

When this parameter is present, the schema for `permission_slug` must be:

```json
{
  "name": "has_permission",
  "in": "query",
  "description": "filter results by permissions authorized to the authenticated principal",
  "required": false,
  "schema": {
    "type": "array",
    "items": {
      "type": "string",
      "enum": [...]
    }
  },
  "style": "form",
  "explode": false
}
```
```
